tests/lib/rules/no-hooks-from-ancestor-modules.js

```diff
@@ -0,0 +1,186 @@
+/**
+ * @fileoverview forbid the use of hooks from ancestor modules
+ * @author Raymond Cohen
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../../lib/rules/no-hooks-from-ancestor-modules"),
+    RuleTester = require("eslint").RuleTester;
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+function createError({ invokedMethodName, expectedHookIdentifierName, usedHookIdentifierName, containingModuleDescription }) {
+    return {
+        messageId: "noHooksFromAncestorModules",
+        data: {
+            invokedMethodName,
+            expectedHookIdentifierName,
+            usedHookIdentifierName,
+            containingModuleDescription
+        }
+    };
+}
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester();
+ruleTester.run("no-hooks-from-ancestor-modules", rule, {
+
+    valid: [
+        "QUnit.testDone(function() {});",
+        `
+        QUnit.module("module", function(hooks) { hooks.beforeEach(function() {}); });
```

Comment:
I need to duplicate all these tests for direct `module()` usage without `QUnit.`

Reply:
Added one valid and one invalid example with direct use of `module`. Let me know if you think more are worth adding.